File: accsaber-database/src/main/java/de/ixsen/accsaber/database/views/extenders/WithRankAndWeightedAp.java
package de.ixsen.accsaber.database.views.extenders;

import org.hibernate.annotations.Immutable;

import javax.persistence.MappedSuperclass;

@MappedSuperclass
@Immutable
public abstract class WithRankAndWeightedAp extends WithRank {

    private Double weightedAp;

    public Double getWeightedAp() {
        return this.weightedAp;
    }
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/views/extenders/WithRankAndWeightedAp.kt
package de.ixsen.accsaber.database.views.extenders

import org.hibernate.annotations.Immutable
import javax.persistence.MappedSuperclass

@MappedSuperclass
@Immutable
abstract class WithRankAndWeightedAp : WithRank() {
    val weightedAp: Double? = null
}